airflow/jobs/local_task_job.py

```diff
@@ -104,11 +104,6 @@ def signal_handler(signum, frame):
         try:
             self.task_runner.start()
 
-            # Unmap the task _after_ it has forked/execed. (This is a bit of a kludge, but if we unmap before
```

Comment:
Unless I am mistaken, I believe this was removed by mistake (and the next piece of code as well). Those two (unmap task and handling signals have been added recently so likely ths is caused by starting off from previous version and not realising the code is removed when rebasing.

Reply:
Welll. I think this is not really good idea to remove the fork. Correct me if I am wrong, but this will ALWAYS add overhead for starting the process and launching a new interpreter, where `fork` was supposed to speed it up immensely. And IMHO it has nothing to do with where DAGs are parsed.

While it might not matter for your case where you have huge DAGs to parse and "fork" vs. "launch process" overhead is negligible, it will significantly change the characteristics and overhead in case the DAGs are small (because the whole interpreter will need to be launched in order to run the task. This impacts both memory and time and I think we should not remove this optimization.

With fork, we use the fact that forking costs almost nothing - both in terms of time and memory. Forked processes use copy-on-write semantics for memory used, which means that most of the dynamic memory allocated before forking will not be re-allocated again in the fork (unless the memory changes).

Unless I missed something - this is a huge change and we should not remove it.

Reply:
cc: @ashb @uranusjr @malthe  -> the usuall summoning here for your comments.

Reply:
BTW. my comment is slightly misplaced, but it concerns the whole PR and removing fork so I will leave it here (but it relates to a number of changes in this PR that remove "fork" option when starting task.

Reply:
> FWIW I just finished implementing (as in a few hours ago) unmapping a deserialised operator in #22396 so this may be irralevant now.

Yeah. I think this is more relevant to removal of the forking in general :) (misplaced comment sorry).

Reply:
@potiuk i have updated the pr description to reflect why the `fork` needs to be removed.  

1. `airflow tasks run --local` process no longer parses dag files. So forking this process won't let the `airflow tasks run --raw` get the parsed dag object,  unless we add additional logic to parse dag file in the fork. 
2. The TaskRunner only forks the `airflow tasks run --local` process when there is no run_as_user. 

I agree that fork can speed up `airflow tasks run --raw` process. I can add it back in this PR so that we can focus mainly on the main topic of AIP 45